### DISCUSSION TITLE
git-tui considerations

### DISCUSSION THREAD

**Discussion by jtrv** [GitoxideLabs/gitoxide#444] [Ideas]
I am starting this discussion to start community input on the git-tui 

[Lazygit](https://github.com/jesseduffield/lazygit) has seemed to be the most ergonomic of the version control TUI's IME. I think if lazygit has not been considered yet, it might be great inspiration for a 'git-tui'.

Also I think it might be worth looking into a different name for 'git-tui' so as not to be confused with [gitui](https://github.com/extrawurst/gitui)... maybe 'gixui' ?

**Comment by Byron** [comment]
Just to clear this up a little, the whole topic around `git-tui` is a bit of a mess. The reason for that is that I don't totally know yet how it should be used. There is another crate, `git-tix`, which definitely will become a simple 'what I always wanted' `tig`-like user interface, mostly for viewing git repositories.

`git-tui` was imagined to be a library crate full of building blocks to create the user interface experience for `ein`, such that `ein` can drive typical user journeys with ease. Despite having never used it, I liked the ideas behind `magit` as it seems to make text actionable. You see it, you can manipulate it, which seems like an efficient approach. The 'typical' TUIs I never really liked and thought there was a lot of room for improvement, so my motivation of studying what's there is mostly to learn which patterns to avoid.

> Also I think it might be worth looking into a different name for 'git-tui' so as not to be confused with [gitui](https://github.com/extrawurst/gitui)... maybe 'gixui' ?

That's a fair point, but I don't think it's going to be an issue as it's a library crate. Future user interfaces will be made available through `ein` directly, and the `tix` program will be provided via the `git-tix` crate`, hence one won't be able to run `cargo install git-tui`.

Judging by my roadmap, I don't think any GUI work will happen this year, or at least it's not planned, and I do wonder when I get to it as using `tig` or the current git user interfaces isn't exactly my biggest gripe. `tix` has the highest chance of being hacked together in a weekend though, maybe if `tig` annoys me enough with its sluggishness in big repositories, it could happen this year.

Besides that, please don't hesitate to drop all of your ideas here, that can't ever hurt.

**Comment by kalkin** [comment]
@JacobTravers Just a shameless plug: There is also https://github.com/kalkin/git-log-viewer
